tests/helpers/mock-ui.js

```diff
@@ -5,16 +5,23 @@ var through = require('through');
 var Promise = require('../../lib/ext/promise');
 
 module.exports = MockUI;
-function MockUI() {
+function MockUI(options) {
   this.output = '';
   this.errors = '';
+  this.errorLog = options && options.errorLog || [];
 
   UI.call(this, {
     inputStream: through(),
     outputStream: through(function(data) {
+      if (options && options.outputStream) {
+        options.outputStream.push(data);
+      }
       this.output += data;
     }.bind(this)),
     errorStream: through(function(data) {
+      if (options && options.errorStream) {
+        options.errorStream.push(data);
```

Comment:
@stefanpenner it's this line that's holding things up. I'm going to look through and make sure that I actually need it, and if not, I'll just remove it.